v2/api/compute/v1api20220702/disk_encryption_set_types_gen.go

```diff
@@ -0,0 +1,2481 @@
+	err = status.AssignProperties_From_DiskEncryptionSet_STATUS(&source.Status)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_From_DiskEncryptionSet_STATUS() to populate field Status")
+	}
+	encryptionSet.Status = status
+
+	// No error
+	return nil
+}
+
+// AssignProperties_To_DiskEncryptionSet populates the provided destination DiskEncryptionSet from our DiskEncryptionSet
+func (encryptionSet *DiskEncryptionSet) AssignProperties_To_DiskEncryptionSet(destination *v20220702s.DiskEncryptionSet) error {
+
+	// ObjectMeta
+	destination.ObjectMeta = *encryptionSet.ObjectMeta.DeepCopy()
+
+	// Spec
+	var spec v20220702s.DiskEncryptionSet_Spec
+	err := encryptionSet.Spec.AssignProperties_To_DiskEncryptionSet_Spec(&spec)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_To_DiskEncryptionSet_Spec() to populate field Spec")
+	}
+	destination.Spec = spec
+
+	// Status
+	var status v20220702s.DiskEncryptionSet_STATUS
+	err = encryptionSet.Status.AssignProperties_To_DiskEncryptionSet_STATUS(&status)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_To_DiskEncryptionSet_STATUS() to populate field Status")
+	}
+	destination.Status = status
+
+	// No error
+	return nil
+}
+
+// OriginalGVK returns a GroupValueKind for the original API version used to create the resource
+func (encryptionSet *DiskEncryptionSet) OriginalGVK() *schema.GroupVersionKind {
+	return &schema.GroupVersionKind{
+		Group:   GroupVersion.Group,
+		Version: encryptionSet.Spec.OriginalVersion(),
+		Kind:    "DiskEncryptionSet",
+	}
+}
+
+// +kubebuilder:object:root=true
+// Generator information:
+// - Generated from: /compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-07-02/diskEncryptionSet.json
+// - ARM URI: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskEncryptionSets/{diskEncryptionSetName}
+type DiskEncryptionSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DiskEncryptionSet `json:"items"`
+}
+
+// +kubebuilder:validation:Enum={"2022-07-02"}
+type APIVersion string
+
+const APIVersion_Value = APIVersion("2022-07-02")
+
+type DiskEncryptionSet_Spec struct {
+	// ActiveKey: The key vault key which is currently used by this disk encryption set.
+	ActiveKey *KeyForDiskEncryptionSet `json:"activeKey,omitempty"`
+
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	AzureName string `json:"azureName,omitempty"`
+
+	// EncryptionType: The type of key used to encrypt the data of the disk.
+	EncryptionType *DiskEncryptionSetType `json:"encryptionType,omitempty"`
+
+	// FederatedClientId: Multi-tenant application client id to access key vault in a different tenant. Setting the value to
+	// 'None' will clear the property.
+	FederatedClientId *string `json:"federatedClientId,omitempty"`
```

Comment:
Should this optionally support coming from a configmap?

Reply:
Thanks for spotting. Done!